### PR TITLE
vimPlugins.nvim-compe: deprecate for nvim-cmp

### PR DESCRIPTION
**Pull request by khaneliman**
## Things done

While going through all the plugins and requires testing. I noticed the nvim-compe plugin has been archived for a couple years and all development moved to nvim-cmp. 

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -2604,7 +2579,8 @@ in
   };
 
   tmux-complete-vim = super.tmux-complete-vim.overrideAttrs {
-    dependencies = with self; [ nvim-compe ];
+    # Vim plugin with optional nvim-compe lua module
+    doCheck = false;
```
Comment (khaneliman) [inline pkgs/applications/editors/vim/plugins/overrides.nix:2583]: Technically does nothing until https://github.com/NixOS/nixpkgs/pull/352277 is implemented.

**Review by mrcjkb — APPROVED** [review]
I was surprised to see nvim-compe extensions in your other PR :sweat_smile: 
LGTM.

File: pkgs/applications/editors/vim/plugins/deprecated.json [hunk 2]
```diff
@@ -43,6 +59,10 @@
         "date": "2021-08-22",
         "new": "bufferline-nvim"
     },
+    "nvim-compe": {
+        "date": "2022-01-27",
```
Comment (m1dugh) [inline pkgs/applications/editors/vim/plugins/deprecated.json:63]: Shouldn't it be today's date ?

**Comment by teto** [comment]
not ruling out there are still users out there (I see it mentioned sometimes on neovim chat), who I encourage to test other solutions.

**Pull request merged**